src/PkgAuthentication.jl

```diff
@@ -309,15 +407,28 @@ ClaimToken immediately, or to Failure if there was an unexpected failure.
 """
 struct RequestLogin <: State
     server::String
+    auth_suffix::String
     challenge::String
-    response::String
+    response::Union{String, Dict{String, Any}}
```

Comment:
Feels like it would have been cleaner to bifurcate the state here to a separate `RequestLoginDevice` state. Then we wouldn't have to overload the same state and handling method with logic from both flows. But not going to insist a change in this PR.